### ISSUE TITLE
Add Parking Location Choice Model for ARC

### ISSUE THREAD

**Issue by danielsclint** [ActivitySim/activitysim#346]
ARC prototype model requires parking location choice model. New model needs to be added to support ARC's (and others') parking location choice implementations.

**Comment by danielsclint** [comment]
@bstabler: You can assign this to me. I'm working on it on my [fork](https://github.com/danielsclint/activitysim/tree/ft_parking_choice). I'm also planning, **at this point**, to add a "psuedo-" parking location choice model to the existing MTC example as my test case.

**Comment by danielsclint** [comment]
I finally got everything working. I went down a few different rabbit holes before I finally found a solution I liked. This solution pivots largely from the trip_destination_choice model with some simplifications removing unnecessary components for the ARC implementation.

Their are two commits with the components:
- [Working Parking Location Choice Model](https://github.com/danielsclint/activitysim/commit/dfdb3faefcda95d0162b7596fbaa656f75ce1fe1): This works! I temporarily slid it into the MTC example for expediency, but I'll separate it once the core functionality has been reviewed. It could probably use some more error checks. Once our analyst completes the trip_mode_choice UEC conversion, we will flush out a more complete model specification from the ARC UEC.
- [Trip Matrix Write](https://github.com/danielsclint/activitysim/commit/fe5a5d9ac5f31f8f632348129ff8dbf318f8263e): Simple addition that looks for the parking location model in the model workflow. If the parking location model is found, it replaces the trip destination with the trip parking zone.

It would be great if @toliwaga and / or @bstabler could take a look and offer feedback.
